api/src/main/java/io/opentelemetry/internal/StringUtils.java

```diff
@@ -57,5 +60,55 @@ public static boolean isValidMetricName(String metricName) {
     return metricName.matches(pattern);
   }
 
+  /**
+   * If given attribute is of type STRING and has more characters than given {@code limit} then
+   * return new AttributeValue with string truncated to {@code limit} characters.
+   *
+   * <p>If given attribute is of type STRING_ARRAY and non-empty then return new AttributeValue with
+   * every element truncated to {@code limit} characters.
+   *
+   * <p>Otherwise return given {@code value}
+   *
+   * @throws IllegalArgumentException if limit is zero or negative
+   */
+  public static AttributeValue cutIfNeeded(AttributeValue value, int limit) {
+    Utils.checkArgument(limit > 0, "attribute value limit cannot be %d", limit);
```

Comment:
recommend including why it failed, not just that the value is invalid. maybe "attribute value limit must be positive"